src/commands/explain.rs

```diff
@@ -51,50 +58,47 @@ fn generate_after_help() -> String {
     format!("{}\n\n{}", list_all_rules(), list_all_tags())
 }
 
-/// Lists all rules as a string for displaying.
-pub fn list_all_rules() -> String {
-    let mut result = String::from("Available rules:");
-    let analysis_rules = analysis::rules();
-    let lint_rules = lint::rules();
-
-    let mut indexes = (0..(analysis_rules.len() + lint_rules.len())).collect::<Vec<_>>();
+/// Gets all rule IDs sorted alphabetically.
+pub fn all_rule_ids() -> Vec<String> {
+    let mut ids: Vec<String> = analysis::rules()
+        .iter()
+        .map(|r| r.id().to_string())
+        .collect();
+    ids.extend(lint::rules().iter().map(|r| r.id().to_string()));
+    ids.sort();
+    ids
+}
 
-    let id = |index: usize| {
-        if index >= analysis_rules.len() {
-            lint_rules[index - analysis_rules.len()].id()
-        } else {
-            analysis_rules[index].id()
+/// Gets all tag names sorted alphabetically.
+fn all_tag_names() -> Vec<String> {
+    let mut tags: HashSet<Tag> = HashSet::new();
```

Comment:
Can we initialize this only once? We don't need to pay the cost of compiling the list every time we call `all_tag_names()`. Same for `all_rule_ids()`.